### PR TITLE
CVSL-2200 Adding in task lists based on status

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -18,7 +18,7 @@ val integrationTest = task<Test>("integrationTest") {
 tasks.register<Copy>("installLocalGitHook") {
   from(File(rootProject.rootDir, ".scripts/pre-commit"))
   into(File(rootProject.rootDir, ".git/hooks"))
-  fileMode = "755".toInt(radix = 8)
+  filePermissions { unix(755) }
```
Comment (andrewrlee) [inline build.gradle.kts:21]: moving away from deprecated api

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt [hunk 2]
```diff
@@ -187,8 +221,19 @@ enum class AssessmentStatus {
...
+  open fun tasks(): Set<TaskProgress> = emptySet()
 }
 
 fun transitions(vararg states: AssessmentStatus) = states.map { Transition(to = it) }.toSet()
 
 data class Transition(val to: AssessmentStatus)
+
+sealed interface TaskProgress {
+  val task: Task
+  val status: (assessment: Assessment) -> TaskStatus
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt:234]: little bit of support to allow dynamically calculating status of task based on additional data though nothing is using this currently

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentLifecycleService.kt [hunk 3]
```diff
@@ -21,8 +22,8 @@ class AssessmentLifecycleService {
   }
 
   fun eligibilityAnswerSubmitted(assessment: AssessmentWithEligibilityProgress): AssessmentStatus {
-    return when {
-      assessment.isChecksPassed() ->
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentLifecycleService.kt:25]: this was more complicated than it needed to be.. I've removed these functions and provided a single calculateStatus flag

File: build.gradle.kts (hunk 1 above)
Comment (justin-gardiner) [reply build.gradle.kts:21]: Nice, that warning was annoying me :)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/TaskProgress.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.model
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.Task
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.TaskStatus
+
+@Schema(description = "Response object which describes an assessment")
```
Comment (justin-gardiner) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/TaskProgress.kt:7]: I guess this needs updating from "an assessment"

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/TaskProgress.kt]: thanks updated that

**Review by justin-gardiner — APPROVED** [review]
LGTM

**Pull request merged**